### PR TITLE
:construction_worker: enforce import constraints

### PR DESCRIPTION
**Pull request by joerunde**
Working on a little import cleanup on the side, and it turns out it's pretty simple to use `pydeps` to see if you're violating any of our implicit constraints. This encodes those "gabe rules" of:
- The core cannot import the runtime
- The core cannot import the interfaces
- The interfaces cannot import the runtime

### CODE REVIEW

File: caikit/core/__init__.py [hunk 1]
```diff
@@ -20,14 +20,16 @@
...
+# "import core"
+
 # Standard
 # We're filtering (most) warnings for now
 import warnings as _warnings
 
 _warnings.filterwarnings("ignore")
 
 # Local
-# must import toolkit first since we need alog to be set up before it is used
```
Comment (evaline-ju) [inline caikit/core/__init__.py:30]: alog at one point was imported through core.toolkit but no longer

Comment (gabe-l-hart) [reply caikit/core/__init__.py:30]: Yes! This has been a long time coming

File: scripts/check_deps.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+set -e
+
+# Use `pydeps` to make sure we're not importing any packages where we shouldn't be
+pydeps --show-dot --no-show --no-config caikit | grep '\->' > deps.txt
+trap "rm deps.txt" EXIT
+
+if < deps.txt grep -q ".*caikit_runtime.*\->.*caikit_core.*"
+then
+    echo "Fail: The core is importing the runtime!"
```
Comment (gabe-l-hart) [inline scripts/check_deps.sh:11]: Love it!

**Review by gabe-l-hart — APPROVED** [review]
So nice to have this!

**Pull request merged**